src/util.rs

```diff
@@ -80,71 +81,96 @@ impl CommandExecution {
+        let path = args.value_of("config").expect("The config arg should exist").to_owned();
+            canonicalize_cli_path(path)
+    } else {
+        // first, check to see if we have a flake.nix in our current directory
+
+        // traverse upwards until we find hive.nix
+        let mut cur = std::env::current_dir()?;
+        let mut hive_path = None;
+
+        loop {
+            let mut listing = match fs::read_dir(&cur) {
+                Ok(listing) => listing,
+                Err(e) => {
+                    // This can very likely fail in shared environments
+                    // where users aren't able to list /home. It's not
+                    // unexpected.
+                    //
+                    // It may not be immediately obvious to the user that
+                    // we are traversing upwards to find hive.nix.
+                    log::warn!("Could not traverse up ({:?}) to find hive.nix: {}", cur, e);
                     break;
-                }
-
-                match cur.parent() {
-                    Some(parent) => {
-                        cur = parent.to_owned();
-                    }
-                    None => {
-                        break;
+                },
+            };
+
+            let found = listing.find_map(|rdirent| {
+                match rdirent {
+                    Err(e) => Some(Err(e)),
+                    Ok(f) => {
+                        if f.file_name() == "hive.nix" {
+                            Some(Ok(f))
+                        } else {
+                            None
+                        }
                     }
                 }
-            }
+            });
 
-            if hive_path.is_none() {
-                log::error!("Could not find `hive.nix` in {:?} or any parent directory", std::env::current_dir()?);
+            if let Some(rdirent) = found {
+                let dirent = rdirent?;
+                hive_path = Some(dirent.path());
+                break;
             }
 
-            hive_path.unwrap()
+            match cur.parent() {
+                Some(parent) => {
+                    cur = parent.to_owned();
+                }
+                None => {
+                    break;
+                }
+            }
         }
-        _ => {
-            let path = args.value_of("config").expect("The config arg should exist").to_owned();
-            canonicalize_cli_path(path)
+
+        if hive_path.is_none() {
+            log::error!("Could not find `hive.nix` in {:?} or any parent directory", std::env::current_dir()?);
         }
+
+        hive_path.unwrap()
     };
 
-    let mut hive = Hive::new(path)?;
+    return Ok(HivePath::from_file(&path));
+}
+
+
+pub fn hive_from_args(args: &ArgMatches<'_>) -> NixResult<Hive> {
+    let path = hive_path_from_args(args)?;
+
+    // grab any extra arguments to pass to Nix
+    let extra_args: Vec<String>;
+    if env::var("COLMENA_NIX_ARGS").is_ok() {
```

Comment:
As it stands `COLMENA_NIX_ARGS` will only be used during evaluation and not deployment. Should it be `COLMENA_EVAL_ARGS`? How can we make adding custom arguments more ergonomic? :thinking: 

I'm also thinking about decoupling Nix command handling into a separate module so Hive and Deployment only interact with clean Rust APIs and we have more flexibility in dealing with special configurations.

Reply:
For now I think I'll just change it to `COLMENA_EVAL_ARGS`, if that's all good with you.